File: practice/easy/odd.py
"""
 Problem: Odd
 URL: http://www.codechef.com/problems/DCE05

"""

__author__ = "Ronald Kaiser"
__email__ = "raios dot catodicos at gmail dot com"


from math import log, floor
   
n = int(raw_input())
for i in range(n):
    print 1 << int(floor(log(int(raw_input()), 2)))

File: practice/old_solutions/odd.c
/*
 Problem: Odd
 URL: http://www.codechef.com/problems/DCE05
 
 Author: Ronald Kaiser
 Email: raios dot catodicos at gmail dot com
*/


#include<stdio.h>
#include<math.h>

int main() {
    int t, n, i, sol;
 
    scanf("%d", &t);
    for(i = 0; i < t; i++) {
        scanf("%d", &n);
        printf("%d\n", 1 << (int)floor(log(n)/log(2)));
    }

    return 0;
}
